### DISCUSSION TITLE
[2023-10-23] GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8136] [Weekly Demo Day]
* Implemented #8109

![image](https://github.com/enso-org/enso/assets/3919101/bfd6f25f-31a8-4060-bb6d-44bce87ba949)

* Investigated how to do visualization previews in the Component Browser better than in the old GUI. Had a call with engine where I described what's needed from the LS API.

**Comment by wdanilo** [comment]
Website progress:


https://github.com/enso-org/enso/assets/1623053/f2783112-4349-405a-b4c5-8461681cf543

We are almost there, but still few things missing :( We had a meeting where we went trough all website sections and estimated what is missing and how much time is still needed. I created initial deployment of the website here: https://enso-org.github.io/website2 (caution: use Chrome to see it and note that fonts are not loaded correctly). Things to be done:

1. Missing animation sections:
   - Collapsing nodes and sharing them with other people (in progress, today will be done)
   - Coding in Python (1 day needed)
   - Working in Sandbox (Environment contexts) (~1 day needed)
   - Running multiple workflows and monitoring their progress (~1 day needed)
   - Connections between nodes (~0.5-1 day needed)
3. Pricing page (only to be created, content is already done) (I will work with E-Hern to code it, design will be done for tomorrow).
4. Integrating chat that E-Hern created (this is almost done, PR is ready, styling to be done).  (I will work with E-Hern to style it, design will be done for tomorrow).
5. Fixing the website for different screen sizes – currently it looks good full-screen on desktop. (I don't know how much it will take, but doesn't look very big).
6. Being sure it works well across browsers – currently Safari has some issues. (I don't know how much it will take, but doesn't look very big).
7. Fixing the deployment – currently, fonts are not loaded correctly. (E-Hern is working on it right now).
8. Some visual fixes (e.g. too long nodes or bad spacings between arguments) (~ 1 day needed).

**Reply by vitvakatu** [reply]
Not sure if it is a known issue, but when I quickly scroll the page up and down, some visual artifacts appear (not 100% of the time, but a green node flying was the most popular one):

https://github.com/enso-org/enso/assets/6566674/41b7c7ab-0dbd-4100-bda3-5bdd870e0db3

**Reply by wdanilo** [reply]
Thanks for catching it, yes, there are many such artifacts. I will be tracking and killing them soon!

**Comment by somebody1234** [comment]
- #8112 (Algorithm only, no UI)
- #8098:

https://github.com/enso-org/enso/assets/4046547/cd369141-069e-4204-b255-016d10f510cd

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/8118 The Documentation panel is on review.
- Clicking links in the documentation panel in progress: https://github.com/enso-org/enso/issues/7992


https://github.com/enso-org/enso/assets/6566674/ae499389-486f-4de5-8060-0a6c947dc5e6

**Comment by kazcw** [comment]
Implemented #7864, PR ready shortly:

[vokoscreenNG-2023-10-23_06-35-31.webm](https://github.com/enso-org/enso/assets/1047859/c1a2af76-b9e4-4ad0-8439-b39ce5f7cec5)

**Comment by MichaelMauderer** [comment]
Working on 
* #7786 (still in progress)

Wip demo: 

[Peek 2023-10-23 14-36.webm](https://github.com/enso-org/enso/assets/1428930/1ff28e6c-4b14-4a25-a549-21b53de08b54)

**Comment by mwu-tow** [comment]
* Still working on https://github.com/enso-org/enso/issues/7863 — expected to have a PR within a day.
* Apple is changing the notarization tool, we need to update our build scripts to use the new one.
* Some CI discussions (incremental builds, random failures)

**Comment by Frizi** [comment]
- Ported graph editor stores and views to Enso AST. The graph now correctly shows only for code from the main function body. PR is waiting for final review. https://github.com/enso-org/enso/pull/8123
- About to start working on widget selection logic.

<img width="459" alt="image" src="https://github.com/enso-org/enso/assets/919491/e0106ee7-aaea-40aa-a42b-fc91c9d8740e">

**Comment by hubertp** [comment]
Relevant backend changes:
- Running Enso parser in the browser #8094 
- Fixed initialization issues in LS that were blocking new GUI #8116 
- Inline execution for better integration with visualizations #8132 
- Investigating capability acquisition problems for new GUI #8133
